recipes/boost/all/conanfile.py

```diff
@@ -958,6 +958,16 @@ def add_defines(library):
         if self.settings.get_safe("compiler.cppstd"):
             flags.append("cxxflags=%s" % cppstd_flag(self.settings))
 
+        if self.settings.get_safe("compiler.sanitizer"):
```

Comment:
`settings.compiler` is not an "official" setting.
Also, it would be weird if only boost would need to listen to this flag. The zlib recipe will ignore `compiler.sanitizer`.

Can you be more specific about what does not work when you add the following to your profile?
```
[env]
CXXFLAGS=-fsanitize=thread
```
Boost has the option `-o boost:debug_level=X` to give more verbose build logs.
Perhaps there you can see what is not working.

Reply:
maybe it's time to refresh old issue: https://github.com/conan-io/conan/issues/1346
as sanitizers are getting more and more popularity in C++ community, I feel like they should get more love from conan

Reply:
Won't conan just add `-fsanitize=xxx` to `CFLAGS/CXXFLAGS/LDFLAGS` when the sanitizers are enabled?
Then we end up in the same situation.

Reply:
I am not sure if it's that simple. some sanitizers may require more flags (e.g. see https://clang.llvm.org/docs/ControlFlowIntegrity.html). or might be incompatible with some flags.
but in 99% of cases, I think you're right, it should just imply `-fsanitize=xxx`.